<a href="https://colab.research.google.com/github/manudsanun/python_dataviz2024/blob/main/Exam3_Manudsanun.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# import ไลบรารี่
import pandas as pd

In [2]:
# ดึงข้อมูลจาก google drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# จากข้อมูลที่ดึงจาก post ที่มีการแชร์เกี่ยวกับประสบการณ์โรคมะเร็งใน Facebook ให้แสดงวิธีการจัดการข้อมูลและนำเสนอจุดที่น่าสนใจในข้อมูลนี้มา 2 จุด

In [4]:
# นำเข้าข้อมูล
data = '/content/drive/MyDrive/Data_Docter_ศ.นพ.ชวลิต.csv'
df = pd.read_csv(data)
df

name  \
0     ศ.นพ.ชวลิต ศูนย์มะเร็ง รพ.จุฬาลงกรณ์   
1                           Nubeer Kithiya   
2                          วาสนา คุ้มจิตต์   
3                     Wansook Chompoo Ploy   
4                              Aphit Aphit   
...                                    ...   
1648                           Fear Q'bick   
1649                          Toei Nittiya   
1650                       Ouiaii Promsuan   
1651                      Thanyalak Banboo   
1652                         Maii Mutchima   

                                               comments  ยอดไลค์  \
0     กดที่ลิงค์นี้เพื่อเข้าไปอ่านคอมเม้นของโพสต์ครั...      456   
1     เป็นมะเร็งลำไส้ใหญ่ ระยะที่ 3 ตอนอายุ 26 ค่ะ พ...     1400   
2     แม่เป็นมะเร็งขั้วปอด ลามขึ้นสมองค่ะ เริ่มแรกที...      402   
3     #มะเร็งปากมดลูกระยะCINlll เกือบเข้าระยะ 1 ตอนอ...     1600   
4     ลูกสาวเป็นมะเร็งกระดูก Osteosarcomaช่วงเข่า อา...      480   
...                                                 ...      ...   
1648            Jitladda Thonglukkeaw มาแชเร้ว ได้บุนนะ        0   
1649                                     Kamta Auttakan        0   
1650                                        อนุชา มีลาภ        0   
1651  Na Nat Fari Kawa II Herbara Binti Abdullah Ras...        0   
1652                              Sasipha Toomsoongnern        0   

      จำนวนการตอบกลับ  
0                   3  
1                  70  
2                  11  
3                  44  
4                   5  
...               ...  
1648                0  
1649                0  
1650                0  
1651                0  
1652                1  

[1653 rows x 4 columns]

In [5]:
# ลบข้อมูลที่มีค่า NAN ออกจากตาราง
df1 = df.dropna()
df1

name  \
0     ศ.นพ.ชวลิต ศูนย์มะเร็ง รพ.จุฬาลงกรณ์   
1                           Nubeer Kithiya   
2                          วาสนา คุ้มจิตต์   
3                     Wansook Chompoo Ploy   
4                              Aphit Aphit   
...                                    ...   
1648                           Fear Q'bick   
1649                          Toei Nittiya   
1650                       Ouiaii Promsuan   
1651                      Thanyalak Banboo   
1652                         Maii Mutchima   

                                               comments  ยอดไลค์  \
0     กดที่ลิงค์นี้เพื่อเข้าไปอ่านคอมเม้นของโพสต์ครั...      456   
1     เป็นมะเร็งลำไส้ใหญ่ ระยะที่ 3 ตอนอายุ 26 ค่ะ พ...     1400   
2     แม่เป็นมะเร็งขั้วปอด ลามขึ้นสมองค่ะ เริ่มแรกที...      402   
3     #มะเร็งปากมดลูกระยะCINlll เกือบเข้าระยะ 1 ตอนอ...     1600   
4     ลูกสาวเป็นมะเร็งกระดูก Osteosarcomaช่วงเข่า อา...      480   
...                                                 ...      ...   
1648            Jitladda Thonglukkeaw มาแชเร้ว ได้บุนนะ        0   
1649                                     Kamta Auttakan        0   
1650                                        อนุชา มีลาภ        0   
1651  Na Nat Fari Kawa II Herbara Binti Abdullah Ras...        0   
1652                              Sasipha Toomsoongnern        0   

      จำนวนการตอบกลับ  
0                   3  
1                  70  
2                  11  
3                  44  
4                   5  
...               ...  
1648                0  
1649                0  
1650                0  
1651                0  
1652                1  

[1653 rows x 4 columns]

## วิเคราะห์ข้อมูลโดยใช้ Interquartile Range (IQR)

IGR การวิเคราะห์ข้อมูลมีความแม่นยำมากขึ้นโดยไม่ถูกบิดเบือนจากข้อมูลที่ผิดปกติ ช่วยประเมินการกระจายตัวของข้อมูลโดยไม่ต้องพึ่งพาการสมมติว่าข้อมูลต้องมีการแจกแจงปกติ ซึ่งแตกต่างจากการใช้ค่าเฉลี่ยและส่วนเบี่ยงเบนมาตรฐานที่อาจถูกบิดเบือนได้หากข้อมูลมีการกระจายตัวที่ไม่สมมาตร

### จัดการข้อมูล

In [32]:
# เลือกเฉพาะที่มีคำว่า "มะเร็ง"
df2= df1[df1['comments'].str.contains('มะเร็ง')]
df2

name  \
1            Nubeer Kithiya   
2           วาสนา คุ้มจิตต์   
3      Wansook Chompoo Ploy   
4               Aphit Aphit   
5                Ize Atitya   
...                     ...   
1639              Amika Phs   
1641  Woranaphatchon Phupho   
1643        Narisa Mohthong   
1644          Chanisara Hom   
1646        Aummy Saenluang   

                                               comments  ยอดไลค์  \
1     เป็นมะเร็งลำไส้ใหญ่ ระยะที่ 3 ตอนอายุ 26 ค่ะ พ...     1400   
2     แม่เป็นมะเร็งขั้วปอด ลามขึ้นสมองค่ะ เริ่มแรกที...      402   
3     #มะเร็งปากมดลูกระยะCINlll เกือบเข้าระยะ 1 ตอนอ...     1600   
4     ลูกสาวเป็นมะเร็งกระดูก Osteosarcomaช่วงเข่า อา...      480   
5     ลูกสาว 5 ขวบ เป็นมะเร็งสมองคะ ไม่มีอาการอะไร น...     1500   
...                                                 ...      ...   
1639  แม่เป็นมะเร็งตับค่ะ เพราะก่อนหน้านั้นแม่ทำงานร...        0   
1641  ไม่ได้เป็นเองแต่ขอเล่าของคุณตาค่ะ เป็นมะเร็งตอ...        0   
1643  พ่อเป็น มะเร็งปอด ระยะ4 (ยังไม่แพร่กระจาย)แบบไ...        0   
1644  พ่อเป็นมะเร็งลำไส้ชอบกินอาหารสุกๆ ดิบค่ะ ขับถ่...        0   
1646  ลุงเป็นมะเร็งหลอดอาหาร แกชอบกินของร้อนๆ ลวกๆคอ...        0   

      จำนวนการตอบกลับ  
1                  70  
2                  11  
3                  44  
4                   5  
5                  23  
...               ...  
1639                0  
1641                0  
1643                0  
1644                0  
1646                0  

[548 rows x 4 columns]

In [43]:
# เลือกข้อมูลที่มีจำนวนตัวอักษรมากกว่า 300 เนื่องจากเป็นการแชร์ประสบการณ์ดังนั้นข้อมูลที่ได้ควรจะมีเนื้อหาที่ยาว
df3 = df2[df2['comments'].str.len() >=300]
df3

name  \
2                    วาสนา คุ้มจิตต์   
3               Wansook Chompoo Ploy   
4                        Aphit Aphit   
6                     Pam Prechanont   
7                         Chiroi Mst   
10                     นู๋นิด ขนิษฐา   
17                     Jin Ping Jern   
20                 Nitchapan Boonlai   
26               SiRi Olive Nareenat   
28                          Sara Apo   
30                    องุ่น รสสุคนธ์   
31                       Tiw Rungsie   
35                 Nuchjaree Srisook   
39                         Jira Jane   
43              Ruethairat Orwattana   
44                Paramapron Singlaw   
45                   Nathi Pinthukas   
46                     Neung Lertsak   
53                 กัญณัฏฐ์ จันทร์คำ   
54                       Domy Dunkin   
62                     KW VVilasinee   
92                    Voravit Kanpai   
93                        แว่น พระพร   
100            นรีวัชร์ ธนัฐเดชวัฒน์   
101                 Utchara Intalook   
102                Chuntid Dedruktip   
105  Malinee Mylittleangle Thongdang   
116               Darunee Phonwithun   
120                  ปลายไง ปลายข้าว   
124                   Bas Bassabassy   
127       Ratthakorn Santiparadonkul   

                                              comments  ยอดไลค์  \
2    แม่เป็นมะเร็งขั้วปอด ลามขึ้นสมองค่ะ เริ่มแรกที...      402   
3    #มะเร็งปากมดลูกระยะCINlll เกือบเข้าระยะ 1 ตอนอ...     1600   
4    ลูกสาวเป็นมะเร็งกระดูก Osteosarcomaช่วงเข่า อา...      480   
6    คุณแม่เป็นมะเร็งเม็ดเลือดขาวแบบเฉียบพลันค่ะ คุ...      300   
7    ประจำเดือนมาผิดปกติค่ะ ไปตรวจภายในที่ รพ. หมอว...       45   
10   ขอให้เรื่องราวของพ่อกับพี่ชายเราเป็นวิทยาทานแก...      353   
17   คุณพ่อเราเป็น #มะเร็งหลังโพรงจมูก ค่ะ นับย้อนไ...      210   
20   คุณแม่บุญธรรม มีศักดิ์เป็นป้าของเราเองสายตรง ไ...      191   
26   คุณแม่เรา ทางกรรมพันธุ์ไม่มีใครเป็นมะเร็งเลยสั...       73   
28   พ่อเป็นมะเร็งลำไส้ใหญ่ เคยสูบบุหรี่สมัยวัยรุ่น...       26   
30   คุณพ่อเป็นมะเร็งตับ ก่อนหน้านี้เวลากินอาหารจะช...      165   
31   สำหรับผู้หญิง ที่มีประจำเดือน ที่มามากผิดปกติ ...      138   
35   ตัวเราเป็นมะเร็งต่อมน้ำเหลืองระยะที่ 2 ค่ะ    ...       43   
39   สวัสดีค่ะ ตัวเองเป็นมะเร็งต่อมน้ำเหลืองชนิดโตช...       25   
43   มาหาข้อมูลว่าเสี่ยงมะเร็งตับไหม สูงทุกปีตรวจล่...        0   
44   ปี  2565  สามีในวัย  36 ปีเจอ  โรค  stroke  เฉ...       19   
45   #มะเร็งตับ แม่ตรวจพบว่าเป็นมะเร็งตับระยะสุดท้า...       17   
46   ผมเป็นมะเร็งเม็ดสีผิวหนังชนิดร้ายแรง(Melanoma)...      477   
53   ตัวเองเป็นมะเร็งเต้านมระยะ 2 เมื่อ 2 ปีที่แล้ว...        2   
54   ผมเป็นมะเร็งต่อมน้ำเหลือง วัย27ปี ไม่กินเหล้าส...      152   
62   จะดีมากเลยค่ะ ถ้าผู้ป่วยที่เป็นมะเร็งรักษาหาย ...        0   
92   อายุ39ปีเป็นมะเร็งลำไส้ครับ แรกๆอาการคล้ายโรคก...      116   
93   ของหนุเป็นมะเร็งเต้านม ไม่มีอาการใดๆเลยหรือปวด...       63   
100  ดิฉันเป็นมะเร็งปอดระยะแพร่กระจายค่ะ มีก้อนมะเร...       88   
101  เป็นมะเร็งช่องปากเริ่มแรกเป็นเหมือนร้อนในที่ลิ...       82   
102  - ขออนุญาตเล่าและแชร์ประสบการณ์นะครับ ซึ่งคนใก...        0   
105  มะเร็งต่อมน้ำเหลือง non hoginค่ะ เจอก้อนที่คอด...       39   
116  ตอนปี 65 เราอายุ 35 ปีเป็นมะเร็งเม็ดเลือดขาวชน...       31   
120  มะเร็งในวัย29 #มะเร็งเต้านมลามต่อมน้ำเหลืองระย...       13   
124  แม่เป็นมะเร็งลำไส้ใหญ่ อาการเริ่มแรกคือ ท้องผู...       19   
127  ผมตรวจพบเป็น #มะเร็งไทรอยด์ ระยะที่ 3 (Papilla...       18   

     จำนวนการตอบกลับ  
2                 11  
3                 44  
4                  5  
6                  9  
7                  4  
10                25  
17                 3  
20                 4  
26                 6  
28                 1  
30                19  
31                13  
35                 3  
39                 5  
43                 1  
44                 6  
45                 2  
46                30  
53                 2  
54                 5  
62                 2  
92                27  
93                 5  
100                9  
101               14  
102     

In [44]:
# เลือกเฉพาะข้อมูลที่มียอไลค์มากกว่า 0
df4 = df3[df3['ยอดไลค์'] > 0]
df4

name  \
2                    วาสนา คุ้มจิตต์   
3               Wansook Chompoo Ploy   
4                        Aphit Aphit   
6                     Pam Prechanont   
7                         Chiroi Mst   
10                     นู๋นิด ขนิษฐา   
17                     Jin Ping Jern   
20                 Nitchapan Boonlai   
26               SiRi Olive Nareenat   
28                          Sara Apo   
30                    องุ่น รสสุคนธ์   
31                       Tiw Rungsie   
35                 Nuchjaree Srisook   
39                         Jira Jane   
44                Paramapron Singlaw   
45                   Nathi Pinthukas   
46                     Neung Lertsak   
53                 กัญณัฏฐ์ จันทร์คำ   
54                       Domy Dunkin   
92                    Voravit Kanpai   
93                        แว่น พระพร   
100            นรีวัชร์ ธนัฐเดชวัฒน์   
101                 Utchara Intalook   
105  Malinee Mylittleangle Thongdang   
116               Darunee Phonwithun   
120                  ปลายไง ปลายข้าว   
124                   Bas Bassabassy   
127       Ratthakorn Santiparadonkul   

                                              comments  ยอดไลค์  \
2    แม่เป็นมะเร็งขั้วปอด ลามขึ้นสมองค่ะ เริ่มแรกที...      402   
3    #มะเร็งปากมดลูกระยะCINlll เกือบเข้าระยะ 1 ตอนอ...     1600   
4    ลูกสาวเป็นมะเร็งกระดูก Osteosarcomaช่วงเข่า อา...      480   
6    คุณแม่เป็นมะเร็งเม็ดเลือดขาวแบบเฉียบพลันค่ะ คุ...      300   
7    ประจำเดือนมาผิดปกติค่ะ ไปตรวจภายในที่ รพ. หมอว...       45   
10   ขอให้เรื่องราวของพ่อกับพี่ชายเราเป็นวิทยาทานแก...      353   
17   คุณพ่อเราเป็น #มะเร็งหลังโพรงจมูก ค่ะ นับย้อนไ...      210   
20   คุณแม่บุญธรรม มีศักดิ์เป็นป้าของเราเองสายตรง ไ...      191   
26   คุณแม่เรา ทางกรรมพันธุ์ไม่มีใครเป็นมะเร็งเลยสั...       73   
28   พ่อเป็นมะเร็งลำไส้ใหญ่ เคยสูบบุหรี่สมัยวัยรุ่น...       26   
30   คุณพ่อเป็นมะเร็งตับ ก่อนหน้านี้เวลากินอาหารจะช...      165   
31   สำหรับผู้หญิง ที่มีประจำเดือน ที่มามากผิดปกติ ...      138   
35   ตัวเราเป็นมะเร็งต่อมน้ำเหลืองระยะที่ 2 ค่ะ    ...       43   
39   สวัสดีค่ะ ตัวเองเป็นมะเร็งต่อมน้ำเหลืองชนิดโตช...       25   
44   ปี  2565  สามีในวัย  36 ปีเจอ  โรค  stroke  เฉ...       19   
45   #มะเร็งตับ แม่ตรวจพบว่าเป็นมะเร็งตับระยะสุดท้า...       17   
46   ผมเป็นมะเร็งเม็ดสีผิวหนังชนิดร้ายแรง(Melanoma)...      477   
53   ตัวเองเป็นมะเร็งเต้านมระยะ 2 เมื่อ 2 ปีที่แล้ว...        2   
54   ผมเป็นมะเร็งต่อมน้ำเหลือง วัย27ปี ไม่กินเหล้าส...      152   
92   อายุ39ปีเป็นมะเร็งลำไส้ครับ แรกๆอาการคล้ายโรคก...      116   
93   ของหนุเป็นมะเร็งเต้านม ไม่มีอาการใดๆเลยหรือปวด...       63   
100  ดิฉันเป็นมะเร็งปอดระยะแพร่กระจายค่ะ มีก้อนมะเร...       88   
101  เป็นมะเร็งช่องปากเริ่มแรกเป็นเหมือนร้อนในที่ลิ...       82   
105  มะเร็งต่อมน้ำเหลือง non hoginค่ะ เจอก้อนที่คอด...       39   
116  ตอนปี 65 เราอายุ 35 ปีเป็นมะเร็งเม็ดเลือดขาวชน...       31   
120  มะเร็งในวัย29 #มะเร็งเต้านมลามต่อมน้ำเหลืองระย...       13   
124  แม่เป็นมะเร็งลำไส้ใหญ่ อาการเริ่มแรกคือ ท้องผู...       19   
127  ผมตรวจพบเป็น #มะเร็งไทรอยด์ ระยะที่ 3 (Papilla...       18   

     จำนวนการตอบกลับ  
2                 11  
3                 44  
4                  5  
6                  9  
7                  4  
10                25  
17                 3  
20                 4  
26                 6  
28                 1  
30                19  
31                13  
35                 3  
39                 5  
44                 6  
45                 2  
46                30  
53                 2  
54                 5  
92                27  
93                 5  
100                9  
101               14  
105               16  
116                8  
120                1  
124                5  
127                4

In [45]:
# เรียงข้อมูลจากน้อยไปมาก
df5 = df4.sort_values(by='ยอดไลค์')
df5

name  \
53                 กัญณัฏฐ์ จันทร์คำ   
120                  ปลายไง ปลายข้าว   
45                   Nathi Pinthukas   
127       Ratthakorn Santiparadonkul   
44                Paramapron Singlaw   
124                   Bas Bassabassy   
39                         Jira Jane   
28                          Sara Apo   
116               Darunee Phonwithun   
105  Malinee Mylittleangle Thongdang   
35                 Nuchjaree Srisook   
7                         Chiroi Mst   
93                        แว่น พระพร   
26               SiRi Olive Nareenat   
101                 Utchara Intalook   
100            นรีวัชร์ ธนัฐเดชวัฒน์   
92                    Voravit Kanpai   
31                       Tiw Rungsie   
54                       Domy Dunkin   
30                    องุ่น รสสุคนธ์   
20                 Nitchapan Boonlai   
17                     Jin Ping Jern   
6                     Pam Prechanont   
10                     นู๋นิด ขนิษฐา   
2                    วาสนา คุ้มจิตต์   
46                     Neung Lertsak   
4                        Aphit Aphit   
3               Wansook Chompoo Ploy   

                                              comments  ยอดไลค์  \
53   ตัวเองเป็นมะเร็งเต้านมระยะ 2 เมื่อ 2 ปีที่แล้ว...        2   
120  มะเร็งในวัย29 #มะเร็งเต้านมลามต่อมน้ำเหลืองระย...       13   
45   #มะเร็งตับ แม่ตรวจพบว่าเป็นมะเร็งตับระยะสุดท้า...       17   
127  ผมตรวจพบเป็น #มะเร็งไทรอยด์ ระยะที่ 3 (Papilla...       18   
44   ปี  2565  สามีในวัย  36 ปีเจอ  โรค  stroke  เฉ...       19   
124  แม่เป็นมะเร็งลำไส้ใหญ่ อาการเริ่มแรกคือ ท้องผู...       19   
39   สวัสดีค่ะ ตัวเองเป็นมะเร็งต่อมน้ำเหลืองชนิดโตช...       25   
28   พ่อเป็นมะเร็งลำไส้ใหญ่ เคยสูบบุหรี่สมัยวัยรุ่น...       26   
116  ตอนปี 65 เราอายุ 35 ปีเป็นมะเร็งเม็ดเลือดขาวชน...       31   
105  มะเร็งต่อมน้ำเหลือง non hoginค่ะ เจอก้อนที่คอด...       39   
35   ตัวเราเป็นมะเร็งต่อมน้ำเหลืองระยะที่ 2 ค่ะ    ...       43   
7    ประจำเดือนมาผิดปกติค่ะ ไปตรวจภายในที่ รพ. หมอว...       45   
93   ของหนุเป็นมะเร็งเต้านม ไม่มีอาการใดๆเลยหรือปวด...       63   
26   คุณแม่เรา ทางกรรมพันธุ์ไม่มีใครเป็นมะเร็งเลยสั...       73   
101  เป็นมะเร็งช่องปากเริ่มแรกเป็นเหมือนร้อนในที่ลิ...       82   
100  ดิฉันเป็นมะเร็งปอดระยะแพร่กระจายค่ะ มีก้อนมะเร...       88   
92   อายุ39ปีเป็นมะเร็งลำไส้ครับ แรกๆอาการคล้ายโรคก...      116   
31   สำหรับผู้หญิง ที่มีประจำเดือน ที่มามากผิดปกติ ...      138   
54   ผมเป็นมะเร็งต่อมน้ำเหลือง วัย27ปี ไม่กินเหล้าส...      152   
30   คุณพ่อเป็นมะเร็งตับ ก่อนหน้านี้เวลากินอาหารจะช...      165   
20   คุณแม่บุญธรรม มีศักดิ์เป็นป้าของเราเองสายตรง ไ...      191   
17   คุณพ่อเราเป็น #มะเร็งหลังโพรงจมูก ค่ะ นับย้อนไ...      210   
6    คุณแม่เป็นมะเร็งเม็ดเลือดขาวแบบเฉียบพลันค่ะ คุ...      300   
10   ขอให้เรื่องราวของพ่อกับพี่ชายเราเป็นวิทยาทานแก...      353   
2    แม่เป็นมะเร็งขั้วปอด ลามขึ้นสมองค่ะ เริ่มแรกที...      402   
46   ผมเป็นมะเร็งเม็ดสีผิวหนังชนิดร้ายแรง(Melanoma)...      477   
4    ลูกสาวเป็นมะเร็งกระดูก Osteosarcomaช่วงเข่า อา...      480   
3    #มะเร็งปากมดลูกระยะCINlll เกือบเข้าระยะ 1 ตอนอ...     1600   

     จำนวนการตอบกลับ  
53                 2  
120                1  
45                 2  
127                4  
44                 6  
124                5  
39                 5  
28                 1  
116                8  
105               16  
35                 3  
7                  4  
93                 5  
26                 6  
101               14  
100                9  
92                27  
31                13  
54                 5  
30                19  
20                 4  
17                 3  
6                  9  
10                25  
2                 11  
46                30  
4                  5  
3                 44

In [46]:
# หาค่าไตรมาสที่ 1 (Q1)
Q1 = df4['ยอดไลค์'].quantile(0.25)
Q1

25.75

จะได้ว่าจำนวนยอดไลค์ที่ 25.75 คือมัธยฐานของข้อมูลส่วนแรก(25%)

In [47]:
# หาค่าไตรมาสที่ 3 (Q3)
Q3 = df4['ยอดไลค์'].quantile(0.75)
Q3

195.75

จะได้ว่าจำนวนยอดไลค์ที่ 195.75 คือมัธยฐานของข้อมูลส่วนที่สาม(75%)

In [48]:
# คำนวณ IQR ; IQR = Q3 - Q1
IQR = Q3 - Q1
IQR

170.0

IQR แสดงถึงการกระจายของข้อมูลกลาง 50% โดยไม่รวมค่าที่สุดโต่งหรือ outliers

In [49]:
# การใช้ IQR ในการระบุ outliers ; [Q1 - 1.5IQR, Q3 + 1.5IQR]
lower = Q1 - 1.5 * IQR
upper = Q3 + 1.5 * IQR
print(lower)
print(upper)

-229.25
450.75


จะได้ว่าข้อมูลที่อยู๋นอกเหนือช่วง [-229.25 - 450.75] เป็น outliers

In [50]:
# เลือกเาเฉพาะข้อมูลที่ไม่เป็น outliers
df5 = df4[(df4['ยอดไลค์'] >= lower) & (df4['ยอดไลค์'] <= upper)]
df5

name  \
2                    วาสนา คุ้มจิตต์   
6                     Pam Prechanont   
7                         Chiroi Mst   
10                     นู๋นิด ขนิษฐา   
17                     Jin Ping Jern   
20                 Nitchapan Boonlai   
26               SiRi Olive Nareenat   
28                          Sara Apo   
30                    องุ่น รสสุคนธ์   
31                       Tiw Rungsie   
35                 Nuchjaree Srisook   
39                         Jira Jane   
44                Paramapron Singlaw   
45                   Nathi Pinthukas   
53                 กัญณัฏฐ์ จันทร์คำ   
54                       Domy Dunkin   
92                    Voravit Kanpai   
93                        แว่น พระพร   
100            นรีวัชร์ ธนัฐเดชวัฒน์   
101                 Utchara Intalook   
105  Malinee Mylittleangle Thongdang   
116               Darunee Phonwithun   
120                  ปลายไง ปลายข้าว   
124                   Bas Bassabassy   
127       Ratthakorn Santiparadonkul   

                                              comments  ยอดไลค์  \
2    แม่เป็นมะเร็งขั้วปอด ลามขึ้นสมองค่ะ เริ่มแรกที...      402   
6    คุณแม่เป็นมะเร็งเม็ดเลือดขาวแบบเฉียบพลันค่ะ คุ...      300   
7    ประจำเดือนมาผิดปกติค่ะ ไปตรวจภายในที่ รพ. หมอว...       45   
10   ขอให้เรื่องราวของพ่อกับพี่ชายเราเป็นวิทยาทานแก...      353   
17   คุณพ่อเราเป็น #มะเร็งหลังโพรงจมูก ค่ะ นับย้อนไ...      210   
20   คุณแม่บุญธรรม มีศักดิ์เป็นป้าของเราเองสายตรง ไ...      191   
26   คุณแม่เรา ทางกรรมพันธุ์ไม่มีใครเป็นมะเร็งเลยสั...       73   
28   พ่อเป็นมะเร็งลำไส้ใหญ่ เคยสูบบุหรี่สมัยวัยรุ่น...       26   
30   คุณพ่อเป็นมะเร็งตับ ก่อนหน้านี้เวลากินอาหารจะช...      165   
31   สำหรับผู้หญิง ที่มีประจำเดือน ที่มามากผิดปกติ ...      138   
35   ตัวเราเป็นมะเร็งต่อมน้ำเหลืองระยะที่ 2 ค่ะ    ...       43   
39   สวัสดีค่ะ ตัวเองเป็นมะเร็งต่อมน้ำเหลืองชนิดโตช...       25   
44   ปี  2565  สามีในวัย  36 ปีเจอ  โรค  stroke  เฉ...       19   
45   #มะเร็งตับ แม่ตรวจพบว่าเป็นมะเร็งตับระยะสุดท้า...       17   
53   ตัวเองเป็นมะเร็งเต้านมระยะ 2 เมื่อ 2 ปีที่แล้ว...        2   
54   ผมเป็นมะเร็งต่อมน้ำเหลือง วัย27ปี ไม่กินเหล้าส...      152   
92   อายุ39ปีเป็นมะเร็งลำไส้ครับ แรกๆอาการคล้ายโรคก...      116   
93   ของหนุเป็นมะเร็งเต้านม ไม่มีอาการใดๆเลยหรือปวด...       63   
100  ดิฉันเป็นมะเร็งปอดระยะแพร่กระจายค่ะ มีก้อนมะเร...       88   
101  เป็นมะเร็งช่องปากเริ่มแรกเป็นเหมือนร้อนในที่ลิ...       82   
105  มะเร็งต่อมน้ำเหลือง non hoginค่ะ เจอก้อนที่คอด...       39   
116  ตอนปี 65 เราอายุ 35 ปีเป็นมะเร็งเม็ดเลือดขาวชน...       31   
120  มะเร็งในวัย29 #มะเร็งเต้านมลามต่อมน้ำเหลืองระย...       13   
124  แม่เป็นมะเร็งลำไส้ใหญ่ อาการเริ่มแรกคือ ท้องผู...       19   
127  ผมตรวจพบเป็น #มะเร็งไทรอยด์ ระยะที่ 3 (Papilla...       18   

     จำนวนการตอบกลับ  
2                 11  
6                  9  
7                  4  
10                25  
17                 3  
20                 4  
26                 6  
28                 1  
30                19  
31                13  
35                 3  
39                 5  
44                 6  
45                 2  
53                 2  
54                 5  
92                27  
93                 5  
100                9  
101               14  
105               16  
116                8  
120                1  
124                5  
127                4

In [55]:
# เรียงลำดับข้อมูลที่ตัด outliers ออกแล้วจาก จำนวนการตอบกลับ และ ยอดไลค์
df6 = df5.sort_values(by = ['จำนวนการตอบกลับ', 'ยอดไลค์'],ascending=False)
df6.head(2)

name                                           comments  \
92  Voravit Kanpai  อายุ39ปีเป็นมะเร็งลำไส้ครับ แรกๆอาการคล้ายโรคก...   
10   นู๋นิด ขนิษฐา  ขอให้เรื่องราวของพ่อกับพี่ชายเราเป็นวิทยาทานแก...   

    ยอดไลค์  จำนวนการตอบกลับ  
92      116               27  
10      353               25

In [62]:
# ข้อมูลที่น่าสนใจข้อมูลที่ 1
print(f'ข้อมูลจากคุณ {df6.iloc[0]["name"]}'
      f'\nแสดงความคิดเห็นดังนี้ {df6.iloc[0]["comments"]}')

ข้อมูลจากคุณ Voravit Kanpai
แสดงความคิดเห็นดังนี้ อายุ39ปีเป็นมะเร็งลำไส้ครับ แรกๆอาการคล้ายโรคกระเพาะเป็นๆหายๆกินได้ปกติน้ำหนักไม่ลดแต่ช่วงๆหลังๆถ่ายผิดปกติไปจากเดิมคือถ่ายวันละ3-4ครั้ง สังเกตุอุจจาระมีมูกเลือดบนไปตรวจพบก้อนเนื้อขนาด7cm ผาตัดเอาชิ้นเนื้อไปตรวจเป็นระยะ3ลามไปต่อมน้ำเหลือง1ต่อม โชคดีเจอคุณหมอใจดีใช้เวลาแค่1อาทิตย์หลังจากเข้าตรวจ คุณหมอนัดมาผ่าตัดให้ในวันหยุด ตอนนี้รอรับคีโมครั้งที่6จากทั้งหมด12ครั้ง สู้โว้ย!


In [63]:
# ข้อมูลที่น่าสนใจข้อมูลที่ 2
print(f'ข้อมูลจากคุณ {df6.iloc[1]["name"]}'
      f'\nแสดงความคิดเห็นดังนี้ {df6.iloc[1]["comments"]}')

ข้อมูลจากคุณ นู๋นิด ขนิษฐา
แสดงความคิดเห็นดังนี้ ขอให้เรื่องราวของพ่อกับพี่ชายเราเป็นวิทยาทานแก่ทุกคนนะคะพ่อเราเป็นมะเร็งตับค่ะ แต่เจ้าตัวเข้าใจว่าเป็นโรคกระเพาะ อาการเหมือนโรคกระเพาะ กินแล้วจุกเสียดแน่นท้อง พ่อก็ซื้อยาลดกรดมากินเองตลอด บอกกินแล้วดีขึ้น อยู่มาวันนึงพ่อตัวเหลืองตาเหลือง กินอะไรไม่ลง ท้องบวมและแข็ง พี่ชายเลยพาไปหาหมอ หมอบอกเป็นโรคหัวใจ แต่ดูหมอไม่แน่ใจเท่าไหร่เลยเจาะเลือดไปตรวจอีกรอบ ผลออกมาว่าเป็น #มะเร็งตับระยะสุดท้าย พ่อถามว่าเขาเป็นมะเร็งใช่ไหม เราบอกไม่ใช่ แค่ตับแข็งเอง แต่เหมือนพ่อรู้ว่าใช่แน่ๆ หลังจากรู้ผลจากหมอแค่ 4 วัน พ่อก็เสียเลย เร็วแบบไม่มีเวลาทำใจเลยค่ะสาเหตุคาดว่า ตอนหนุ่มๆพ่อทั้งกินเหล้า ทั้งสูบบุหรี่จัดมาก น่าจะสะสมมานานพ่อเสียได้ 2 ปีกว่า พี่ชายก็เป็นมะเร็งตามพ่อมาติดๆเลย พี่ชายเป็นมะเร็งกระเพาะอาหาร เขาเป็นคนแข็งแรงมากไม่เคยป่วยเลย แต่อยู่ๆก็กินแล้วจุกเสียดแน่นท้อง พี่ก็คิดว่าเป็นโรคกระเพาะ ก็ซื้อยามากินเอง แต่ดูไม่ดีขึ้น แถมแย่ลง พี่ก็เข้าใจว่าเป็นกรดไหลย้อน แต่ก็แย่ลงอย่างรวดเร็ว น้ำหนักลด 13 กิโลในเดือนเดียว พี่ชายไปหาหมอ ที่แรกตรวจไม่เจอ บอกกระเพาะ